File: Array/find_second_maximum.js
function findSecondMaximum(arr) {
    let firstMax = Number.NEGATIVE_INFINITY;
    let secondmax = Number.NEGATIVE_INFINITY;

    for(let i = 0;i<arr.length;i++) {
        if(arr[i] > firstMax) {
            secondmax = firstMax;
            firstMax = arr[i];
        } else if(arr[i] > secondmax && arr[i] !=firstMax) {
            secondmax = arr[i];
        } 
    } 
    return secondmax;
}

console.log(findSecondMaximum([4,5,7,7]));

File: Day05/find_second_maximum.py
def find_second_maximum(lst):
    first_max = float('-inf')
    second_max = float("-inf")

    for item in lst:
        if item > first_max:
            first_max = item
    
    for item in lst:
        if item != first_max and item > second_max:
            second_max = item
    
    return second_max

print(find_second_maximum([9,4,3,5,8]))